src/Symfony/Component/Console/Descriptor/TextDescriptor.php

```diff
@@ -298,7 +298,7 @@ private function formatDefaultValue($default): string
     }
 
     /**
-     * @param (Command|string)[] $commands
+     * @param array<Command|string> $commands
```

Comment:
I don't know about this notation, I think `array<Foo|string>` is widespread in SCA tools. We plan to support this notation in the `DebugClassLoader`. Note that here it does not matter on a `@param` but some `@return` use it and I changed them all for consitency.